docs/rules/style/pointless-reassignment.md

```diff
@@ -0,0 +1,62 @@
+# pointless-reassignment
+
+**Summary**: Pointless reassignment of variable
+
+**Category**: Style
+
+**Avoid**
+```rego
+package policy
+
+allow if {
+    users := all_users
+    any_admin(users)
+}
+```
+
+**Prefer**
+```rego
+package policy
+
+allow if {
+    any_admin(all_users)
+}
+```
+
+## Rationale
+
+Values and variables are immutable in Rego, so reassigning the value of one variable to another only adds noise.
+
+## Exceptions
+
+Reassigning the value of a long reference often helps readability, and especially so when it needs to be referenced
+multiple times:
+
+```rego
+package policy
+
+allow if {
+    users := input.context.permissions.users
+    any_admin(users)
+}
+```
+
+This rule does not consider that a violation.
```

Comment:
```suggestion
This rule does not consider such examples violations.
```

Reply:
I wonder what if I a had a rule like `foo_bar_whizzz_bang_bar_foo_baz_v1_foo`, and then reassigned it for readability, would this be good to ignore too?

Reply:
I'd say the issue then would be you naming a rule or var like that in the first place :D